packages/@aws-cdk/aws-appsync/test/appsync-auth.test.ts

```diff
@@ -0,0 +1,603 @@
+  });
+
+  test('appsync does not create unspecified api key with empty additionalAuthorizationModes', () => {
+    // WHEN
+    new appsync.GraphQLApi(stack, 'api', {
+      name: 'api',
+      schemaDefinition: appsync.SchemaDefinition.FILE,
+      schemaDefinitionFile: path.join(__dirname, 'appsync.test.graphql'),
+      authorizationConfig: {
+        defaultAuthorization: { authorizationType: appsync.AuthorizationType.IAM },
+        additionalAuthorizationModes: [],
+      },
+    });
+
+    // THEN
+    expect(stack).not.toHaveResource('AWS::AppSync::ApiKey');
+  });
+
+  test('appsync creates configured api key with additionalAuthorizationModes', () => {
+    // WHEN
+    new appsync.GraphQLApi(stack, 'api', {
+      name: 'api',
+      schemaDefinition: appsync.SchemaDefinition.FILE,
+      schemaDefinitionFile: path.join(__dirname, 'appsync.test.graphql'),
+      authorizationConfig: {
+        defaultAuthorization: { authorizationType: appsync.AuthorizationType.IAM },
+        additionalAuthorizationModes: [{
+          authorizationType: appsync.AuthorizationType.API_KEY,
+          apiKeyConfig: { description: 'Custom Description' },
+        }],
+      },
+    });
+
+    // THEN
+    expect(stack).toHaveResourceLike('AWS::AppSync::ApiKey', {
+      Description: 'Custom Description',
+    });
+  });
+
+  test('appsync creates configured api key with additionalAuthorizationModes (not as first element)', () => {
+    // WHEN
+    new appsync.GraphQLApi(stack, 'api', {
+      name: 'api',
+      schemaDefinition: appsync.SchemaDefinition.FILE,
+      schemaDefinitionFile: path.join(__dirname, 'appsync.test.graphql'),
+      authorizationConfig: {
+        defaultAuthorization: { authorizationType: appsync.AuthorizationType.IAM },
+        additionalAuthorizationModes: [
+          {
+            authorizationType: appsync.AuthorizationType.USER_POOL,
+            userPoolConfig: { userPool: new cognito.UserPool(stack, 'myPool') },
+          },
+          {
+            authorizationType: appsync.AuthorizationType.API_KEY,
+            apiKeyConfig: { description: 'Custom Description' },
+          },
+        ],
+      },
+    });
+
+    // THEN
+    expect(stack).toHaveResourceLike('AWS::AppSync::ApiKey', {
+      Description: 'Custom Description',
+    });
+  });
+
+  test('appsync fails when empty default and API_KEY in additional', () => {
+    // WHEN
+    const when = () => {
+      new appsync.GraphQLApi(stack, 'api', {
+        name: 'api',
+        schemaDefinition: appsync.SchemaDefinition.FILE,
+        schemaDefinitionFile: path.join(__dirname, 'appsync.test.graphql'),
+        authorizationConfig: {
+          additionalAuthorizationModes: [{
+            authorizationType: appsync.AuthorizationType.API_KEY,
+          }],
+        },
+      });
+    };
+
+    // THEN
+    expect(when).toThrowError('You can\'t duplicate API_KEY configuration. See https://docs.aws.amazon.com/appsync/latest/devguide/security.html');
+  });
```

Comment:
we typically wrap the code that throws with the when

i.e.

```
expect(() => { //code that throws
  }).toThrowError("message");
```

I know this was already existing code, but let's align it with our normal conventions in this chore/refactor?

Reply:
the code is a little big tho.. are you suggesting just putting everything into the expect?